app/client/cli/flags/config.go

```diff
@@ -0,0 +1,19 @@
+package flags
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/pokt-network/pocket/runtime/configs"
+)
+
+var Cfg *configs.Config
+
+func ParseConfigAndFlags(_ *cobra.Command, _ []string) error {
+	// by this time, the config path should be set
+	Cfg = configs.ParseConfig(ConfigPath)
+
+	// set final `remote_cli_url` value; order of precedence: flag > env var > config > default
+	RemoteCLIURL = viper.GetString("remote_cli_url")
```

Comment:
I intended to revert this change. This is the direction I started when investigating https://github.com/pokt-network/pocket/pull/891#discussion_r1263340347, I thought I had reverted it but apparently failed.

To answer your question, the reason for this is because viper's only integration with flags is to support setting a viper key based on the flag, but not the other way around. I.e: Viper won't update the flag value. **This only applies to bound flags**

This means that we have to do one of the following consistently for bound flags:

- call `viper.GetString("<flag key>")` (or a helper containing it) anywhere we need the value
- update the flag consistently and reference the flag value anywhere we need it

I opted for the latter option as I felt it was more conventional and easier to read and maintain.